### PR TITLE
update CI, read specific vars in npzread

### PR DESCRIPTION
**Pull request by jishnub**
Firstly this PR adds the option to read specific variables in `npzread` instead of all the variables.

Secondly it adds docstrings to `npzread` and `npzwrite`. It also adds a documentation generation workflow.

Thirdly it updates the CI to github actions from travis, and updates Tagbot as [suggested here](https://discourse.julialang.org/t/ann-required-updates-to-tagbot-yml/49249). Tbh I'm not totally sure if this is done correctly.

### CODE REVIEW

**Comment by jishnub** [comment]
The readme badge needs to be updated, perhaps this can be done after merging.

**Review by fhs — CHANGES_REQUESTED** [review]
Thanks!

I'm don't know how to the documentation deployment works. From a brief look at the docs, I guess it pushes the generated documentation to this repo?

I would've preferred a separate PR for the `npzread` changes, but it's fine.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '0.7'
+          - '1.0'
+          - '1'
```
[the remark this answers is not in the document]
Comment (jishnub) [reply .github/workflows/ci.yml:20]: My bad, I will update this. Any particular reason to test on 1.1 though, given that it is not supported anymore?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          - 'nightly'
+        os:
+          - ubuntu-latest
```
[the remark this answers is not in the document]
Comment (jishnub) [reply .github/workflows/ci.yml:23]: I will add it, my bad, I had copied the template from PkgCompiler

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jishnub) [reply .github/workflows/ci.yml:20]: Along the same lines, testing on 1.0 should be enough to not require a separate test on 0.7, as 1.0 is essentially 0.7 without deprecation messages

**Comment by jishnub** [comment]
yes I realize now that separate PRs would have been a better idea.

The documentation workflow pushes commits to the `gh-pages` branch. It might be necessary to select this branch as the source of github pages from settings. Although I've only used travis for this, and there might be [issues with using github actions for this](https://github.com/JuliaDocs/Documenter.jl/issues/1177). Reading through the issue it seems that [using an SSH deploy key ](https://github.com/JuliaDocs/Documenter.jl/issues/1177#issuecomment-555034592)should get around this, so I'm adding this for the time being.

File: docs/Project.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[deps]
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+NPZ = "15e1cf62-19b3-5cfa-8e77-841668bca605"
+
+[compat]
+Documenter = "0.25"
```
[the remark this answers is not in the document]
Comment (jishnub) [reply docs/Project.toml]: I've added this now

File: src/NPZ.jl [hunk 4]
```diff
@@ -235,36 +235,64 @@ function samestart(a::AbstractVector, b::AbstractVector)
...
+    if ext == ".npy"
+        name = fname
+    end
+    name
+end
+
+"""
+    npzread(filename::AbstractString, [vars])
+
+Read an variable or a collection of variables from `filename`.
```
[the remark this answers is not in the document]
Comment (jishnub) [reply src/NPZ.jl]: Fixed

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jishnub) [reply .github/workflows/ci.yml]: Fixed

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fhs) [reply .github/workflows/ci.yml:21]: Good to know. I'll change them to currently supported versions, which I think are 1.5 and 1.0.

**Review by fhs — APPROVED** [review]
Thanks!

**Pull request merged**